### PR TITLE
feat: APPS-2774 Create CardMeta component FTVA variation

### PR DESCRIPTION
**Pull request by jendiamond**
Connected to [APPS-2774](https://jira.library.ucla.edu/browse/APPS-2774) | [PR-518](https://github.com/UCLALibrary/ucla-library-website-components/issues/518)

**Component Updated:** `CardMeta.vue`

**Notes:**
The ShareButton is being added as a slot.

The styles will be updated when the UX updates the tokens.

**Checklist:**

-  [x] I checked that it is working locally in the dev server
-  [x] I checked that it is working locally in the storybook
-  [x] I added a screenshot of it working
-  [x] UX has reviewed and approved this
-  [x] I assigned this PR to someone on the dev team to review
-  [x]  I used a conventional commit message
-  [x] I assigned myself to this PR

[APPS-2774]: https://uclalibrary.atlassian.net/browse/APPS-2774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PR-518]: https://uclalibrary.atlassian.net/browse/PR-518?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/lib-components/BlockTag.vue [hunk 1]
```diff
@@ -61,13 +64,25 @@ const parsedClasses = computed(() => {
...
-    <component :is="BlockTagIcons[iconName]" v-if="iconName" class="svg" aria-hidden="true" />
-    <div class="label" v-html="label" />
+    <component
+      :is="BlockTagIcons[iconName]"
+      v-if="iconName"
+      class="svg"
+      aria-hidden="true"
+    />
+
+    <div class="label">
```
Comment (farosFreed) [inline src/lib-components/BlockTag.vue:74]: Why was it necessary to make this change to BlockTag? This breaks the link implementation https://deploy-preview-530--ucla-library-storybook-vue3x.netlify.app/?path=/story/block-tag--html-link-ftva

We need to revert this if possible

**Review by farosFreed — CHANGES_REQUESTED** [review]

File: src/lib-components/CardMeta.vue [hunk 2]
```diff
@@ -107,10 +129,16 @@ const parsedLocations = computed(() => {
     }
   })
 })
+
+// if we want to use theme pattern based on passing classes prop to each component
+const parsedClasses = computed(() => {
```
Comment (pghorpade) [inline src/lib-components/CardMeta.vue:134]: Rename parsedClasses to classes, like other components toplevel computed class method

File: src/lib-components/CardMeta.vue [hunk 3]
```diff
@@ -107,10 +129,16 @@ const parsedLocations = computed(() => {
...
+  // (maybe classes.join() if need an array)
+  return ['card-meta', theme?.value || '', props.isCentered ? 'centered' : '']
+})
 </script>
 
 <template>
-  <div class="card-meta">
+  <div :class="parsedClasses">
```
Comment (pghorpade) [inline src/lib-components/CardMeta.vue:141]: Rename parsedClasses to classes, like other components toplevel computed class method

File: src/lib-components/BlockTag.vue [hunk 4]
```diff
@@ -61,13 +64,24 @@ const parsedClasses = computed(() => {
 
 <template>
   <span :class="parsedClasses">
```
Comment (pghorpade) [inline src/lib-components/BlockTag.vue:66]: Rename parsedClasses to classes, like other components toplevel computed class method

Comment (pghorpade) [inline src/lib-components/BlockTag.vue:66]: Rename parsedClasses to classes, like other components toplevel computed class method

File: src/lib-components/CardMeta.vue [hunk 5]
```diff
@@ -1,20 +1,27 @@
...
+// import BlockTag from '@/lib-components/BlockTag.vue'
 
 // UTILITY FUNCTIONS
 import formatTimes from '@/utils/formatEventTimes'
 import formatDates from '@/utils/formatEventDates'
 
 // import removeHtmlTruncate from '@/utils/removeHtmlTruncate'
 
-import type { LocationItemType } from '@/types/types'
+import type { LocationItemType, ScreeningTagItemType } from '@/types/types'
```
Comment (pghorpade) [inline src/lib-components/CardMeta.vue:24]: what is a ScreeningTagItemType? What naming pattern are we following? Should it be more like what it is in craft, like the craft field itself?

File: src/lib-components/CardMeta.vue [hunk 6]
```diff
@@ -1,20 +1,27 @@
...
 import { computed } from 'vue'
 import { useRoute } from 'vue-router'
 import type { PropType } from 'vue'
+import { useTheme } from '@/composables/useTheme'
 
 // COMPONENTS
 import SmartLink from '@/lib-components/SmartLink.vue'
 import IconWithLink from '@/lib-components/IconWithLink.vue'
 import RichText from '@/lib-components/RichText.vue'
+import BlockTag from '@/lib-components/BlockTag.vue'
```
Comment (pghorpade) [inline src/lib-components/CardMeta.vue:14]: We should lazy load this component as it is not required data.

File: src/lib-components/CardMeta.vue [hunk 7]
```diff
@@ -69,10 +76,25 @@ const props = defineProps({
...
   },
+  tagLabels: {
+    type: Array as PropType<ScreeningTagItemType[]>,
+    default: () => [],
+  },
+  introduction: {
+    type: String,
+    default: '',
+  },
+  isCentered: {
```
Comment (pghorpade) [inline src/lib-components/CardMeta.vue:87]: Along with isCentered, also add useRoute to check if it is a collection home page path if the centered is only on the collection homepage to show centered content, Ask UX if not sure about the route. See getSectionName in DividerWayFinder

File: src/lib-components/CardMeta.vue [hunk 8]
```diff
@@ -183,6 +211,30 @@ const parsedLocations = computed(() => {
         :to="location.to"
       />
     </div>
+
+    <div
+      v-if="tagLabels.length"
```
Comment (pghorpade) [inline src/lib-components/CardMeta.vue:216]: The check should be tagLabels && tagLabels.length > 0

**Review by farosFreed — APPROVED** [review]

File: src/lib-components/CardMeta.vue (hunk 5 above)
Comment (jendiamond) [reply src/lib-components/CardMeta.vue:23]: The craft field is screeningTag

Comment (jendiamond) [reply src/lib-components/CardMeta.vue:23]: Screenshot 2024-07-01 at 10 47 13 PM

File: src/lib-components/CardMeta.vue (hunk 7 above)
Comment (jendiamond) [reply src/lib-components/CardMeta.vue:86]: Update:
the centering is only on landing pages - if a collection has items inside
So we are going to do the centering on the page level.

**Review by pghorpade — APPROVED** [review]
👍

**Review by pghorpade — APPROVED** [review]
👍

**Pull request merged**